### PR TITLE
aria labels and roles

### PR DESCRIPTION
**Pull request by k-cross**
This is to make a blog more a11y friendly by including more labels for things and fewer obstacles for visual elements that don't matter.

### CODE REVIEW

File: templates/_footer.html [hunk 1]
```diff
@@ -1,13 +1,13 @@
-<footer>
-  <div class="copyright">
+<footer aria-label="Footer">
```
Comment (isunjn) [inline templates/_footer.html:1]: unnecessary

File: templates/_header.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
-<header {% if config.extra.blur_effect %} class="blur" {% endif %}>
-  <div id="header-wrapper">
-    <nav>
-      <a href="/">{{ config.extra.id }}</a>
+<header {% if config.extra.blur_effect %} class="blur" {% endif %} aria-label="Main Menu">
```
Comment (isunjn) [inline templates/_header.html:1]: unnecessary

File: templates/_header.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+  <div id="header-wrapper" role="none">
```
Comment (isunjn) [inline templates/_header.html:2]: i think there is no need to add role="none" to a div element

File: templates/_header.html [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
+    <nav class="main-nav" role="none">
```
Comment (isunjn) [inline templates/_header.html:3]: why add role="none" to a nav? the class is unnecessary too

File: templates/_header.html [hunk 5]
```diff
@@ -32,7 +32,7 @@
         <a href="{{ section.path }}">{{ section.name }}</a>
       {% endif %}
     </nav>
-    <div id="btns">
+    <div id="btns" aria-label="Options">
```
Comment (isunjn) [inline templates/_header.html:35]: unnecessary, also i think "options" is a confused word there

File: templates/blog.html [hunk 6]
```diff
@@ -9,16 +9,16 @@
 
 {% block content %}
 {% include "_header.html" %}
-<div id="wrapper">
-  <main class="layout-list">
+<div id="wrapper" role="none">
```
Comment (isunjn) [inline templates/blog.html:12]: no need

File: templates/blog.html [hunk 7]
```diff
@@ -9,16 +9,16 @@
...
+  <main class="layout-list" aria-label="Topics">
```
Comment (isunjn) [inline templates/blog.html:13]: "topics" is a confused word here

File: templates/blog.html [hunk 8]
```diff
@@ -9,16 +9,16 @@
...
     {% if config.extra.blog_categorized %}
     {% for category,posts in section.pages | group_by(attribute="taxonomies.categories.0") -%}
-    <div class="category">{{ category }}</div>
+    <div class="category" aria-label="Topic">{{ category }}</div>
```
Comment (isunjn) [inline templates/blog.html:16]: same

File: templates/home.html [hunk 9]
```diff
@@ -32,27 +32,27 @@
 {% endblock head %}
 
 {% block content %}
-<div id="wrapper">
-  <main>
-    <section id="info">
+<div id="wrapper" role="none">
```
Comment (isunjn) [inline templates/home.html:35]: unnecessary

File: templates/home.html [hunk 10]
```diff
@@ -32,27 +32,27 @@
...
+  <main aria-label="Content Sections">
```
Comment (isunjn) [inline templates/home.html:36]: unnecessary

File: templates/home.html [hunk 11]
```diff
@@ -32,27 +32,27 @@
...
+    <section id="info" aria-label="Info">
       <img src="{{ config.extra.avatar }}" alt="avatar">
-      <div id="text">
-        <div>
-          <span id="name">{{ config.extra.name }}</span>
+      <div id="text" role="none">
```
Comment (isunjn) [inline templates/home.html:39]: unnecessary

File: templates/home.html [hunk 12]
```diff
@@ -32,27 +32,27 @@
...
+        <div role="none">
```
Comment (isunjn) [inline templates/home.html:40]: unnecessary

File: templates/home.html [hunk 13]
```diff
@@ -32,27 +32,27 @@
...
+          <span id="name" aria-label="Name">{{ config.extra.name }}</span>
           {% if config.extra.display_id -%}
-          <span id="id">@{{ config.extra.id }}</span>
+          <span id="id" aria-label="ID">@{{ config.extra.id }}</span>
           {% endif -%}
         </div>
-        <div id="bio"> {{ config.extra.bio }} </div>
+        <div id="bio" aria-label="Blurb"> {{ config.extra.bio }} </div>
```
Comment (isunjn) [inline templates/home.html:46]: why not "bio"?

File: templates/home.html [hunk 14]
```diff
@@ -32,27 +32,27 @@
...
       </div>
     </section>
-    <section id="links">
-      <div id="left">
+    <section id="links" aria-label="Navigation Links">
+      <div id="left" role="none">
```
Comment (isunjn) [inline templates/home.html:50]: unnecessary

File: templates/home.html [hunk 15]
```diff
@@ -32,27 +32,27 @@
...
         {% for section in config.extra.sections %}
         <a href="{{ section.path }}" {% if section.is_external %}target="_blank" rel='noreferrer noopener'{% endif %}>{{ section.name }}</a>
         {% endfor %}
       </div>
-      <div id="right">
+      <div id="right" role="none">
```
Comment (isunjn) [inline templates/home.html:55]: unnecessary

File: templates/home.html [hunk 16]
```diff
@@ -68,7 +68,7 @@
       </div>
     </section>
     {% if config.extra.homepage_layout == 'about' %}
-      <section id="brief" class="prose">
+      <section id="brief" class="prose" aria-label="About">
```
Comment (isunjn) [inline templates/home.html:71]: unnecessary

File: templates/post.html [hunk 17]
```diff
@@ -45,20 +45,20 @@
 
 {% block content %}
 {% include "_header.html" %}
-<div id="wrapper">
-  <div id="blank"></div>
-  <aside {% if config.extra.blur_effect %} class="blur" {% endif %}>
+<div id="wrapper" role="none">
```
Comment (isunjn) [inline templates/post.html:48]: unnecessary

File: templates/post.html [hunk 18]
```diff
@@ -45,20 +45,20 @@
...
+  <div id="blank" aria-hidden=true role="none"></div>
```
Comment (isunjn) [inline templates/post.html:49]: unnecessary

File: templates/post.html [hunk 19]
```diff
@@ -85,10 +85,10 @@
       <div id="copy-cfg" style="display: none;" data-copy-icon="{{ copy_icon }}" data-check-icon="{{ check_icon }}"></div>
       {% endif %}
       {% set backlink_icon = load_data(path="static/icon/backlink.svg") %}
-      <article class="prose" data-backlink-icon="{{ backlink_icon }}">
-        <h1>{{ page.title }}</h1>
-        <div id="post-info">
-          <div id="date">
+      <article class="prose" data-backlink-icon="{{ backlink_icon }}" aria-label="{{ page.title }}">
```
Comment (isunjn) [inline templates/post.html:88]: no need

File: templates/post.html [hunk 20]
```diff
@@ -85,10 +85,10 @@
...
+        <h1 role="none">{{ page.title }}</h1>
```
Comment (isunjn) [inline templates/post.html:89]: why?

File: templates/projects.html [hunk 21]
```diff
@@ -9,20 +9,20 @@
 
 {% block content %}
 {% include "_header.html" %}
-<div id="wrapper">
-  <main>
+<div id="wrapper" role="none">
```
Comment (isunjn) [inline templates/projects.html:12]: unnecessary

File: templates/projects.html [hunk 22]
```diff
@@ -9,20 +9,20 @@
...
+  <main aria-label="Projects">
```
Comment (isunjn) [inline templates/projects.html:13]: unnecessary

File: templates/projects.html [hunk 23]
```diff
@@ -9,20 +9,20 @@
...
-      <div class="more">
-        <div class="tags">
+    <div class="proj" aria-label="Project">
+      <p class="name" aria-label="{{ proj.name }}">{{ proj.name }}</p>
+      <div class="desc" aria-label="Description">{{ proj.desc | trim | markdown | safe }}</div>
+      <div class="more" aria-label="Metadata">
+        <div class="tags" aria-label="Tags">
           {% for tag in proj.tags %}
-          <div><span>#</span>{{ tag }}</div>
+          <div role="none"><span role="none">#</span>{{ tag }}</div>
```
Comment (isunjn) [inline templates/projects.html:22]: unnecessary

**Review by isunjn — CHANGES_REQUESTED** [review]

**Comment by k-cross** [comment]
I'm definitely not an a11y pro but I thought a tree with more contextual information would be better for the vision impaired.

Current:
Screenshot 2023-11-27 at 09 33 13

Branch:
Screenshot 2023-11-27 at 09 36 33

**Comment by isunjn** [comment]
i'm not a pro either, i will take a closer look at a11y topics and then decide what to add, thanks for the pr anyway!

**Comment by isunjn** [comment]
I took a brief look, and it seems like we should not give every node on a11y tree a name/label，what matters is controls, interactive/focusable elements.

The "contextual information" you added might actually be annoying/needless for the vision impaired.

Also the [firefox accessibility inspector](https://firefox-source-docs.mozilla.org/devtools-user/accessibility_inspector/) can tell you the a11y "issues" on your page，and seems the only noticable issue of this theme is color contrast problem, i plan to change the default primary color in future releases

**Comment by k-cross** [comment]
Sorry, I just noticed this in my inbox. Sorry for the noise.

**Pull request closed without merging**